src/main/java/com/connexta/store/controllers/StoreController.java

```diff
@@ -69,19 +72,30 @@
     }
 
     final Long fileSize = file.getSize();
-    // TODO validate that fileSize is (0 GB, 10 GB]
+    if (fileSize > MAX_FILE_BYTES) {
+      throw new CreateDatasetException(
+          BAD_REQUEST,
+          String.format(
+              "File size is %d bytes. File size cannot be greater than %d bytes",
+              fileSize, MAX_FILE_BYTES));
+    }
 
     final String mediaType = file.getContentType();
-    // TODO verify that mediaType is not blank and is a valid Content Type
+    if (mediaType == null) {
+      throw new CreateDatasetException(BAD_REQUEST, "Media type is missing");
+    }
 
     final String fileName = file.getOriginalFilename();
-    // TODO verify that fileName is not blank
+    if (StringUtils.isEmpty(fileName)) {
```

Comment:
❓ Do we need to do any sort of escaping on these inputs?

Reply:
Not that I know of.